src/command.ts

```diff
@@ -255,10 +260,27 @@ export default abstract class Command {
       if (deprecated) {
         this.warn(formatFlagDeprecationWarning(flag, deprecated))
       }
+
+      const deprecateAliases = this.ctor.flags[flag]?.deprecateAliases
+      const aliases = this.ctor.flags[flag]?.aliases ?? []
+      if (deprecateAliases && aliases) {
```

Comment:
`aliases` will always be true since it's `?? []` (empty arrays are truthy).  

Did you maybe want aliases.length?